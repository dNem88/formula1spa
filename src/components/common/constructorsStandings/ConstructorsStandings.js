import React, {useEffect, useState, Fragment} from 'react'
import styles from './constructorsStandings.module.css'
import hStandings from '../../../utils/api/heroku_api/hStandings'
import ErrorComp from '../errorComp/ErrorComp'
import ConstructorsPoints from '../constructorsPoints/ConstructorsPoints'
import Spinner from '../spinner/Spinner'


function ConstructorsStandings() {
    const [standings, setStandings] = useState({
        standings: null,
        hasError: null,
        errorMessage: null
    })
    useEffect(() => {
        async function FetchStandings() {
            try{
                let response = await hStandings.getConstructorsStandings()
                if (!response.ok) {
                    throw new Error('Bad request to Standings')
                }
                let json = await response.json()
                if (json.error) {
                    throw new Error(json.error.message)
                }
                setStandings({standings: json, hasError: null, errorMessage: null})
                return json
            }catch(err) {
                setStandings({standings: null, hasError: true, errorMessage: err.message})
                return err
            }
        }
        FetchStandings()
        
    }, [])

    return (
        <Fragment>
             {(standings.hasError) ? <ErrorComp errorMessage={standings.errorMessage}/> : 
                <div className={styles['main-container']}>
                    < div className = {
                        styles['heading-container']
                    } >
                        <h2>CONSTRUCTORS STANDINGS</h2>
                    </div>
                    <div className={styles['standings-container']}>
                        {!standings.standings ? <Spinner/> : 
                            standings.standings.map(x => {
                                return <ConstructorsPoints key={x.constructor} {...x}/>
                            })
                        }
                    </div>
                </div>
             }
        </Fragment>
    )
}

export default ConstructorsStandings