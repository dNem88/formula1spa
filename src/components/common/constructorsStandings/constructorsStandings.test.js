import React from 'react'
import renderer from 'react-test-renderer'
import ConstructorsStandings from './ConstructorsStandings'
import {MemoryRouter} from 'react-router-dom'



jest.mock('../constructorsPoints/ConstructorsPoints', () => () => 'Constructor Poitns Mock')

describe('Testing Constructor Standings Component', () => {
    it('render properly ', () => {
        const snapshot = renderer.create(
            <MemoryRouter>
                <ConstructorsStandings/>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('div')
    })

    
})