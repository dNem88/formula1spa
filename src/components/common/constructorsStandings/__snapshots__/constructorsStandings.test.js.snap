// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Constructor Standings Component render properly  1`] = `
<div
  className="main-container"
>
  <div
    className="heading-container"
  >
    <h2>
      CONSTRUCTORS STANDINGS
    </h2>
  </div>
  <div
    className="standings-container"
  >
    <div
      className="lds-ring"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
